SAMZA-2596: Replace String.format() calls to avoid MissingFormatArgumentException

Zhangyx39 opened the pull request:
Symptom: String.format("..." + key) would through MissingFormatArgumentException if the key string contains something like "%3A" or "%d". And original exception would be hidden by this new exception.
 
Cause: We have no control on what the key can be. Should avoid using String.format().
 
Changes: Simply remove the String.format() should be enough.
 
Tests: No test since it's a minor change.

API Changes: No API change.

Upgrade Instructions: No backward-compatibility issue. No need to upgrade.

Usage Instructions: User won't use this code directly.

lakshmi-manasa-g reviewed (commented):
Changes look good. 
Wondering about a few things
1. are the other changes (Collections.emptyMap()), not casting to CompletableFuture<T> and   "(throwable) e" are necessary for this change?
2. Does String.format cause a problem in other places? is it restricted only to Table api. What about CachingTable?

Zhangyx39 commented:
@lakshmi-manasa-g Thanks for the review!
1. They are not related to this change. I saw some warnings on IDE about them. So I updated as well. These changes should have no effect.
2. We only had this issue in table API. Good suggestion. Also updated for CachingTable.

lakshmi-manasa-g reviewed (approved):
thank you for addressing CachingTable.
lgtm.

sborya reviewed (approved):
looks good
